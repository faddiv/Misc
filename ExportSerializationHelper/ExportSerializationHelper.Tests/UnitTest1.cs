using Bogus;
using CsvHelper;
using CsvHelper.Configuration;
using DocumentFormat.OpenXml.Packaging;
using FluentAssertions;
using Snapshooter.Xunit;
using System;
using System.Globalization;
using System.IO;
using System.Text;
using Xunit;

namespace ExportSerializationHelper.Tests
{
    public class UnitTest1
    {
        [Fact]
        public void Test1()
        {
            var today = DateTime.Parse("2022-03-06");
            var config = new ExampleClassMap();
            var faker = new Faker<ExampleClass>()
                .UseSeed(1234)
                .RuleFor(e => e.Id, f => f.Random.Number(1000))
                .RuleFor(e => e.TextValue, f => f.Random.AlphaNumeric(10))
                .RuleFor(e => e.TimestampValue, f => f.Date.Recent(refDate: today))
                .RuleFor(e => e.NullableDate, f => f.Random.Bool() ? f.Date.Recent(refDate: today) : null);
            var data = faker.Generate(10);
            using var stream = new MemoryStream();
            using var writer = new StreamWriter(stream);
            using var csv = new CsvWriter(writer, new CsvConfiguration(CultureInfo.InvariantCulture)
            {
                HasHeaderRecord = false
            });

            foreach (var header in config.GetHeader())
            {
                csv.WriteField(header);
            }
            csv.NextRecord();

            foreach (var item in ExportSerializer.Serialize(data, config))
            {
                foreach (var field in item.Data)
                {
                    csv.WriteField(field);
                }
                csv.NextRecord();
            }
            csv.Flush();
            var result = writer.Encoding.GetString(stream.ToArray());

            result.Should().MatchSnapshot();
        }

        [Fact]
        public void Test2()
        {
            var ctl = new SamplesController();
            var mem = ctl.CreateExcelFile();
            File.WriteAllBytes("excel.xlsx", mem.ToArray());
        }

    }
}
