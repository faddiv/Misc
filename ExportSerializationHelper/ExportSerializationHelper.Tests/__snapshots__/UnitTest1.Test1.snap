"SWRlbnRpZmllcixTb21lIFRleHQsVGltZXN0YW1wLE51bGxhYmxlDQozOTksd2J5Y3l0aW5iZiwwMy8wNS8yMDIyIDAzOjA4OjQzLA0KNDY0LHFvbDIyNHNsajQsMDMvMDUvMjAyMiAwODozOTo1MSwwMy8wNS8yMDIyIDAyOjQ3OjAzDQoyODQsZXdyaGl4ajg3YywwMy8wNS8yMDIyIDA3OjU4OjA1LA0KODQyLGo1eGlpNnNwYnIsMDMvMDUvMjAyMiAwMjo0ODoyMSwwMy8wNS8yMDIyIDAyOjQyOjQ1DQoyNzgsc2M2YjQyaTRxaywwMy8wNS8yMDIyIDAxOjU1OjA5LDAzLzA1LzIwMjIgMjM6Mjg6NTYNCjkzNixpaDhlZ2E4Z21sLDAzLzA1LzIwMjIgMjA6NDY6MTUsMDMvMDUvMjAyMiAyMTo0MzowNg0KNTgzLDI0djJtbnF3MDEsMDMvMDUvMjAyMiAxMDo1ODozMiwNCjMxOCxmZWNsNTV1N3VqLDAzLzA1LzIwMjIgMTI6MzM6NDQsDQo2MDQsMWNxbDYyeXgxbSwwMy8wNS8yMDIyIDA3OjI1OjIwLA0KNzIwLGQ3bWc2dnd5czEsMDMvMDUvMjAyMiAxNzozMzoxOSwwMy8wNS8yMDIyIDAwOjI3OjMwDQo="
